corehq/messaging/scheduling/views.py

```diff
@@ -107,6 +117,130 @@ def get(self, request, *args, **kwargs):
         return super(BroadcastListView, self).get(*args, **kwargs)
 
 
+class CreateMessageView(BaseMessagingSectionView):
+    urlname = 'create_message'
+    page_title = ugettext_lazy('Create a Message')
+    template_name = 'scheduling/create_message.html'
+
+    @method_decorator(_requires_new_reminder_framework())
+    @method_decorator(requires_privilege_with_fallback(privileges.OUTBOUND_SMS))
+    @use_select2
+    def dispatch(self, *args, **kwargs):
+        return super(CreateMessageView, self).dispatch(*args, **kwargs)
+
+    @cached_property
+    def project_timezone(self):
+        return get_timezone_for_user(None, self.domain)
+
+    @property
+    def parent_pages(self):
+        return [
+            {
+                'title': BroadcastListView.page_title,
+                'url': reverse(BroadcastListView.urlname, args=[self.domain]),
+            },
+        ]
+
+    @property
+    def form_kwargs(self):
+        return {
+            'domain': self.domain,
+        }
+
+    @cached_property
+    def message_form(self):
+        if self.request.method == 'POST':
+            return MessageForm(self.request.POST, **self.form_kwargs)
+        return MessageForm(**self.form_kwargs)
+
+    @property
+    def page_context(self):
+        return {
+            'form': self.message_form,
+        }
+
+    @cached_property
+    def project_languages(self):
+        doc = StandaloneTranslationDoc.get_obj(self.domain, 'sms')
+        return getattr(doc, 'langs', ['en'])
+
+    def post(self, request, *args, **kwargs):
+        if self.message_form.is_valid():
+            # TODO editing should not create a new one
+            values = self.message_form.cleaned_data
+            if values['send_frequency'] == 'immediately':
+                if values['translate']:
+                    messages = {}
+                    for lang in self.project_languages:
+                        key = 'message_%s' % lang
+                        if key in values:
+                            messages[lang] = values[key]
+                    content = SMSContent(message=messages)
+                else:
+                    content = SMSContent(message={'*': values['non_translated_message']})
+                schedule = AlertSchedule.create_simple_alert(self.domain, content)
+                broadcast = ImmediateBroadcast(
+                    domain=self.domain, name=values['schedule_name'], schedule=schedule
+                )
+                broadcast.save()
```

Comment:
would be good to edit the schedule and broadcast in a transaction

Reply:
https://github.com/dimagi/commcare-hq/pull/17722/commits/6f3806dd1c4b2be72ffe1ba40404abf50314c719